src/ast/mod.rs

```diff
@@ -1124,6 +1124,14 @@ pub enum Expr {
     /// [Databricks](https://docs.databricks.com/en/sql/language-manual/sql-ref-lambda-functions.html)
     /// [DuckDb](https://duckdb.org/docs/sql/functions/lambda.html)
     Lambda(LambdaFunction),
+    /// Checks membership of a value in a JSON array
+    ///
+    /// Syntax:
+    /// ```sql
+    /// <value> MEMBER OF(<array>)
+    /// ```
+    /// [MySQL](https://dev.mysql.com/doc/refman/8.4/en/json-search-functions.html#operator_member-of)
+    MemberOf(Box<Expr>, Box<Expr>),
```

Comment:
Can we introduce a struct for the expression e.g. `MemberOf(MemberOf{ value, json })`? That way it would be less breaking if spans or other options are added to the expression later on